### ISSUE TITLE
Evaluate download chunk size

### ISSUE THREAD

**Issue by johnbradley** [Imageomics/pyMorphoSource#20]
Right now when downloading files we use a chunk size of 128:
https://github.com/Imageomics/pyMorphoSource/blob/f34dec314fae5f579aa8650c93cab782999ef4ff/src/morphosource/download.py#L44
Make this option configurable from `download_media_bundle()` and see if we can get a more performant default value.

**Comment by johnbradley** [comment]
I did some testing to compare different chunks sizes from python and also using curl directly for downloading.
I downloaded the same 34.6 MB file 5 times and bars below are the average with different chunk size/tool.

## Downloading on Laptop over WiFi
<img width="607" alt="Screenshot 2023-12-11 at 9 38 32 AM" src="https://github.com/Imageomics/pyMorphoSource/assets/1024463/de9d45dd-79ae-43b4-8805-658e4efbb9c6">

## Downloading on OSC
<img width="606" alt="Screenshot 2023-12-11 at 9 38 41 AM" src="https://github.com/Imageomics/pyMorphoSource/assets/1024463/71dbcfa9-abd9-4f98-be2a-96de4054b263">

## TLDR
The current 128 chunk size doesn't seem to make a large difference in time to download.
1KB might be a slightly better value.

**Comment by muratmaga** [comment]
how do these compare to the download speeds of the same file from the website?

**Comment by johnbradley** [comment]
I manually downloaded the file (media id 000390204) via chrome 5x and updated the chart with a **browser** column:
<img width="611" alt="Screenshot 2023-12-11 at 10 27 37 AM" src="https://github.com/Imageomics/pyMorphoSource/assets/1024463/e4be4c3b-6f29-4fa5-870f-bece3c8d44bb">
The browser measurement was using the stopwatch on my phone so it's not as precise as the other measurements.

SlicerMorph may have different download performance than these metrics because it has custom logic for downloading (QT progress bar and threading).

**Comment by muratmaga** [comment]
Thanks. Your browser speeds seems still too slow, and might be impeding on your test. For large files (>1GB), we are getting download speeds of 50-100MB/sec from the web browser on the JS2, and that's what Oshane is going to test. The specific file media ID we are using is 000016055. Morphosource also rotates files based on access frequency (cold storage vs hot), and maybe caching things. So best to stick to the same file of reasonable size (i,e, representative of typical microCT datasets).
@oothomas please post your results here.

**Comment by johnbradley** [comment]
For larger files (>1GB) I have found that large chunk sizes do help with downloading speed. In a previous project we used 20MB chunk size and it helped with larger files. I spoke with @oothomas, and he will be running a test with the SlicerMorph code on Jetstream with this larger chunk size. Once that test is done I'll test with my simpler pyMorphoSource only test script using the larger file above on Jetstream.

**Comment by johnbradley** [comment]
Results from JetStream with the larger file:
<img width="607" alt="Screenshot 2023-12-11 at 4 27 29 PM" src="https://github.com/Imageomics/pyMorphoSource/assets/1024463/de979450-84c1-4601-aba0-2694f5e6be9e">
